bin/knc.c

```diff
@@ -955,7 +959,9 @@ write_network_buffer(work_t *work)
 		    &(work->network_buffer.out[work->network_buffer.out_pos]),
 		    work->network_buffer.out_len);
 
-	if (len < 0 && ((errno != EINTR) && (errno != EAGAIN))) {
+	if (len < 0) {
+		if (errno == EINTR || errno == EAGAIN)
+			return 1; /* retry later (do not return <= 0) */
```

Comment:
The comment block for this function needs to be updated.  Callers don't treat `1` as "fully transmitted" but as "no error".